### PR TITLE
MINOR: Refactor configs in GroupMetadataManager

### PR DESCRIPTION
**Pull request by dajac**
This patch does a few things:
1) It refactors the GroupMetadataManager to received a GroupCoordinatorConfig object instead of having to pass all the configs individually. It simplifies adding new configs.
2) It replaces usages of `Collections.singletonList/Map` by `List.of/Map.of`.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: server/src/main/java/org/apache/kafka/server/config/AbstractKafkaConfig.java [hunk 1]
```diff
@@ -43,31 +43,32 @@
  */
 public abstract class AbstractKafkaConfig extends AbstractConfig {
     @SuppressWarnings("deprecation")
```
Comment (chia7712) [inline server/src/main/java/org/apache/kafka/server/config/AbstractKafkaConfig.java]: Could you please remove this unused `SuppressWarnings` also?

File: group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorConfig.java [hunk 2]
```diff
@@ -367,6 +383,26 @@ public GroupCoordinatorConfig(AbstractConfig config) {
                 SHARE_GROUP_HEARTBEAT_INTERVAL_MS_CONFIG, SHARE_GROUP_SESSION_TIMEOUT_MS_CONFIG));
     }
 
+    @SuppressWarnings("unchecked")
+    private static List<ConsumerGroupPartitionAssignor> consumerGroupAssignors(
+        AbstractConfig config
+    ) {
+        // In unit tests, it is pretty convenient to have the ability to pass instantiated
```
Comment (chia7712) [inline group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorConfig.java]: Yes, using the same object is indeed convenient for testing. However, it still acts as a backdoor. Perhaps we could shift this behavior out of production and into the `GroupMetadataManagerTestContext`, allowing it to override the `GroupCoordinatorConfig` to return the object directly. What do you think?

File: group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java [hunk 3]
```diff
@@ -1905,7 +1718,7 @@ private CoordinatorResult<ConsumerGroupHeartbeatResponseData, CoordinatorRecord>
                 metrics.record(CONSUMER_GROUP_REBALANCES_SENSOR_NAME);
             }
 
-            group.setMetadataRefreshDeadline(currentTimeMs + consumerGroupMetadataRefreshIntervalMs, groupEpoch);
+            group.setMetadataRefreshDeadline(currentTimeMs + Integer.MAX_VALUE, groupEpoch);
```
Comment (chia7712) [inline group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java]: It seems the "deadline" is a no-op due to `Integer.MAX_VALUE`, correct?

**Review by chia7712 — COMMENTED** [review]
@dajac this is indeed a great refactor. some minor comments are left. PTAL

File: group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorConfig.java (hunk 2 above)
Comment (dajac) [reply group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorConfig.java]: Let me see what I can do.

File: group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java (hunk 3 above)
Comment (dajac) [reply group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java]: Correct. We don't rely on periodic refreshed at the moment. Let me still extract it into a constant.

File: group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorConfig.java (hunk 2 above)
Comment (dajac) [reply group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorConfig.java]: I just pushed a change. Let me know what you think.

**Review by chia7712 — APPROVED** [review]

File: group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java [hunk 4]
```diff
@@ -5252,7 +5066,7 @@ private CoordinatorResult<Void, CoordinatorRecord> tryCompleteInitialRebalanceEl
         if (group.newMemberAdded() && remainingMs != 0) {
             // A new member was added. Extend the delay.
             group.setNewMemberAdded(false);
-            int newDelayMs = Math.min(classicGroupInitialRebalanceDelayMs, remainingMs);
+            int newDelayMs = Math.min(config.classicGroupInitialRebalanceDelayMs(), remainingMs);
             int newRemainingMs = Math.max(remainingMs - delayMs, 0);
```
Comment (chia7712) [inline group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java:5070]: @dajac Given that `newDelayMs` is the value utilized by the scheduler, should we modify the expression to `Math.max(remainingMs - newDelayMs, 0)`?

**Pull request merged**